lib/auditd/auditd_test.go

```diff
@@ -0,0 +1,280 @@
+//go:build linux
+
+/*
+ *
+ * Copyright 2022 Gravitational, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package auditd
+
+import (
+	"bytes"
+	"encoding/binary"
+	"sync"
+	"testing"
+
+	"github.com/mdlayher/netlink"
+	"github.com/mdlayher/netlink/nlenc"
+	"github.com/stretchr/testify/require"
+)
+
+type msgOrErr struct {
+	msg netlink.Message
+	err error
```

Comment:
it looks like you never actually use the err field

Reply:
Added a test case that uses it. Thanks.